test/test_restart.py

```diff
@@ -0,0 +1,159 @@
+# Copyright (c) 2021 elParaguayo
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+import pickle
+import shutil
+import textwrap
+
+import pytest
+
+import libqtile.bar
+import libqtile.config
+import libqtile.layout
+import libqtile.resources.default_config
+from libqtile.ipc import IPCError
+from libqtile.widget import TextBox
+from test.helpers import TestManager as BareManager
+
+
+class MinimalConf(libqtile.confreader.Config):
+    auto_fullscreen = False
+    keys = []
+    mouse = []
+    groups = [
+        libqtile.config.Group("a"),
+        libqtile.config.Group("b"),
+        libqtile.config.Group("c"),
+        libqtile.config.Group("d")
+    ]
+    layouts = [
+        libqtile.layout.Stack(num_stacks=1),
+        libqtile.layout.Max()
+    ]
+    floating_layout = libqtile.resources.default_config.floating_layout
+    fake_screens = [
+        libqtile.config.Screen(
+            top=libqtile.bar.Bar([TextBox("Qtile Test")], 10),
+            x=0, y=0, width=400, height=600
+        ),
+        libqtile.config.Screen(
+            top=libqtile.bar.Bar([TextBox("Qtile Test")], 10),
+            x=400, y=0, width=400, height=600
+        )
+    ]
+    screens = []
```

Comment:
What parts of this are needed for the tests in this module, and which are there because they have to? Perhaps it'd be neater to import `BareConfig` from `helpers` and just explicitly have those defined here that are used here?

Reply:
Yes, that would be neater.